server/models/Collective.js

```diff
@@ -2409,6 +2409,28 @@ function defineModel() {
       });
     }
 
+    await sequelize.query(
+      `
+        UPDATE public."Orders" SET status = 'CANCELLED'
+        WHERE id IN (
+          SELECT "Orders".id FROM public."Orders"
+          INNER JOIN public."Subscriptions" ON "Subscriptions".id = "Orders"."SubscriptionId"
+          WHERE
+            "Orders".status = 'ACTIVE' AND
+            "Subscriptions"."isManagedExternally" AND
+            "Subscriptions"."isActive" AND
+            "Orders"."CollectiveId" = ?
+        )
+      `,
+      {
+        type: QueryTypes.UPDATE,
+        replacements: [this.id],
+      },
+    );
+
+    const virtualCards = await models.VirtualCard.findAll({ where: { CollectiveId: this.id } });
+    await Promise.all(virtualCards.map(virtualCard => virtualCard.destroy()));
```

Comment:
:ice_cream: Personal pref because many collectives don't have many virtual cards attached, but an alternative is:

```suggestion
    await models.VirtualCard.destroy({ where: { CollectiveId: this.id } });
```

Reply:
This should actually have been `virtualCard.delete()`, thanks for pointing this out.
The delete method in the VirtualCard model deletes the card with the provider before deleting from the database